cli/azd/resources/scaffold/templates/resources.bicept

```diff
@@ -459,14 +477,32 @@ module {{bicepName .Name}} 'br/public:avm/res/app/container-app:0.8.0' = {
           {{- end}}
           {{- if (and .AzureStorageAccount .AzureStorageAccount.AuthUsingManagedIdentity) }}
           {
-            name: 'SPRING_CLOUD_AZURE_STORAGE_CREDENTIAL_MANAGEDIDENTITYENABLED'
+            name: 'SPRING_CLOUD_AZURE_EVENTHUBS_PROCESSOR_CHECKPOINTSTORE_CONNECTIONSTRING'
```

Comment:
FYI: `SPRING_CLOUD_AZURE_EVENTHUBS_PROCESSOR_CHECKPOINTSTORE_CONNECTION_STRING` can not work.
Reason:

> ![image](https://github.com/user-attachments/assets/5ed881b9-e77a-45af-8364-3cd73d1186be)

Convert all or convert none are supported. Convert part is not supported.
1. Convert all: `SPRING_CLOUD_AZURE_EVENTHUBS_PROCESSOR_CHECKPOINT_STORE_CONNECTION_STRING`.
2. Convert none: `SPRING_CLOUD_AZURE_EVENTHUBS_PROCESSOR_CHECKPOINTSTORE_CONNECTIONSTRING`.
3. Convert some: `SPRING_CLOUD_AZURE_EVENTHUBS_PROCESSOR_CHECKPOINTSTORE_CONNECTIO_NSTRING`.

Reply:
I am just curious how they match the `CHECKPOINT_STORE` here, is it that if it's a camel case, such as `aB`, it will switch to `A_B`?

Reply:
For `aB`, it will firstly convert to `a-b`, then supports both `AB` and `A_B`,

Another example: For `aBcD` it will firstly convert to `a-bc-d`, then surrprts both `ABCD` and `A_BC_D`, but not support others like `A_BCD`. The suggested way is using `ABCD`. 

Refs:
1. https://docs.spring.io/spring-boot/reference/features/external-config.html#features.external-config.typesafe-configuration-properties.relaxed-binding
> ![image](https://github.com/user-attachments/assets/65ad7c62-96f1-4035-8c21-65d999554e5f)


2. https://github.com/spring-projects/spring-boot/blob/6470748d6ddc32c041196c2c0a05983232a9cc1f/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/SystemEnvironmentPropertyMapper.java#L79-L80

![image](https://github.com/user-attachments/assets/282056c1-bbba-4c29-94df-8dec7d7c0a8f)